src/dali/plugin/input/csv/blockfi.py

```diff
@@ -133,9 +177,98 @@ def load(self) -> List[AbstractTransaction]:
                 else:
-                    self.__logger.debug("Unsupported transaction type (skipping): %s", raw_data)
+                    self.__logger.error("Unsupported transaction type (skipping): %s", raw_data)
+
+        if self.__trade_csv_file:
+            result += self.parse_trade_report(self.__trade_csv_file)
+        return result
+
+    def parse_trade_report(self, file_path: str) -> List[AbstractTransaction]:
+        result: List[AbstractTransaction] = []
+
+        with open(file_path, mode="r", encoding="utf-8") as csv_file:
+            lines = reader(csv_file)
+            # skip csv header
+            header = next(lines)
+            self.__logger.debug("Header: %s", header)
+
+            col_idx = {}
+            for (idx, name) in enumerate(header):
+                col_idx[name] = idx
+
+            for line in lines:
+                raw_data: str = self.__DELIMITER.join(line)
+                self.__logger.debug("Transaction: %s", raw_data)
+
+                trans_type = line[col_idx["Type"]]
+                if trans_type != "Trade":
+                    raise Exception(f"Internal error: unsupported transaction type: {trans_type}")
+
+                trade_id = line[col_idx["Trade ID"]]
+                date = line[col_idx["Date"]]
+                timestamp = f"{date} -00:00"
+                from_currency = line[col_idx["Sold Currency"]].upper()
+                from_size = line[col_idx["Sold Quantity"]]
+                to_currency = line[col_idx["Buy Currency"]].upper()
+                to_size = line[col_idx["Buy Quantity"]]
+
+                result.append(
+                    OutTransaction(
+                        plugin=self.__BLOCKFI,
+                        unique_id=trade_id,
+                        raw_data=raw_data,
+                        timestamp=timestamp,
+                        asset=from_currency,
+                        exchange=self.__BLOCKFI,
+                        holder=self.account_holder,
+                        transaction_type="Sell",
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_out_no_fee=from_size,
+                        crypto_fee="0",
+                        notes=f"Sell side of trade: {from_size} {from_currency} -> {to_size} {to_currency}",
+                    ),
+                )
+                result.append(
+                    InTransaction(
+                        plugin=self.__BLOCKFI,
+                        unique_id=f"{trade_id}/buy",
+                        raw_data=raw_data,
+                        timestamp=timestamp,
+                        asset=to_currency,
+                        exchange=self.__BLOCKFI,
+                        holder=self.account_holder,
+                        transaction_type="Buy",
+                        spot_price=Keyword.UNKNOWN.value,
+                        crypto_in=to_size,
+                        fiat_fee="0",
```

Comment:
This looks like a crypto -> crypto conversion: in this case doesn't BlockFi charge a crypto fee? If so it can be captured by using the crypto_fee of InTransaction (introduced very recently in both DaLI and RP2): see CoinbasePro coin swap for an example.

Reply:
blockfi doesn't charge crypto fees

Reply:
Oh, really? I wish Coinbase did the same :-D (they charge handsomely).

Reply:
blockfi hides the fee through higher premiums :face_exhaling: at least coinbase is being transparent :)